src/cluster.c

```diff
@@ -5617,7 +5663,12 @@ NULL
         /* CLUSTER FORGET <NODE ID> */
         clusterNode *n = clusterLookupNode(c->argv[2]->ptr, sdslen(c->argv[2]->ptr));
         if (!n) {
-            addReplyErrorFormat(c,"Unknown node %s", (char*)c->argv[2]->ptr);
+            if (clusterBlacklistExists((char*)c->argv[2]->ptr))
+                /* Already forgotten. The deletion may have been gossipped by
+                 * another node, so we pretend it succeeded. */
+                addReply(c,shared.ok);
+            else
+                addReplyErrorFormat(c,"Unknown node %s", (char*)c->argv[2]->ptr);
             return;
         } else if (n == myself) {
```

Comment:
@zuiderkwast what do you think about the idea of allowing the node to be deleted to initiate the gossip message as well? This would save another explicit operation on the admin side.

Reply:
Would this be a new command? Like `CLUSTER LEAVE` or something?

Seems like a good idea. A node could broadcast this message repeatedly until everyone responds with a message that it has been blacklisted.

Reply:
Sure, especially if it makes sense with cluster V2 as well. But let's do that in another PR...